modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/IngestCommonPlugin.java

```diff
@@ -82,6 +82,7 @@ public IngestCommonPlugin() {
         processors.put(KeyValueProcessor.TYPE, new KeyValueProcessor.Factory());
         processors.put(URLDecodeProcessor.TYPE, new URLDecodeProcessor.Factory());
         processors.put(BytesProcessor.TYPE, new BytesProcessor.Factory());
+        processors.put(ConditionalProcessor.TYPE, new ConditionalProcessor.Factory(parameters.scriptService));
```

Comment:
I thought we were only exposing the conditional via each processor as `if`, not as a processor on its own?

Reply:
See https://github.com/elastic/elasticsearch/pull/32398#issuecomment-408132873 and discussion leading up to it. This is just an implementation to prevent us from having to adjust every processor.

Reply:
Sorry I don't understand. Can you reiterate? I don't understand why using it from parsing `if` requires it be generally available. As it is here, it would be available to construct directly by a user, right?

Reply:
@rjernst 

Sure, let me try to make it short:
In order to get the conditional we have to either add a conditional evaluation to every `execute` method for every processor or make some abstract parent have an `execute` implementation that holds the conditional evaluation. (I mean there's other options, but I think the two mentioned are the simplest and others will be even more noisy/risky/...)
Either option requires us to change all existing processors and also their factories (as a result of us parsing the configuration in each processor factory).
=> I went for this implementation since it was shortest and didn't have any functional/performance downsides over alternatives anyway (as far as I can see).

> As it is here, it would be available to construct directly by a user, right?

Yea true, if you think that's a problem I can prevent that in the parser easily though :) Should I?

Reply:
Yes I think we should prevent that. But I'm still not understanding why it is necessary for this processor to have a factory or be registered. It can be constructed completely locally, and directly via its ctor, within `ConfigurationUtils`.

Reply:
@rjernst that's more of a convenience thing because `org.elasticsearch.ingest.ConfigurationUtils#readProcessor(java.util.Map<java.lang.String,org.elasticsearch.ingest.Processor.Factory>, java.lang.String, java.lang.Object)` (which is called from like 3 places in prod. code) would to also get the script factory as an input then (which will trigger a pretty big change if you factor in test code).

Just tried to keep this less noisy again :) => bad idea?, better to add the script factory as an input here?

Reply:
With what I'm suggesting, the script factory is not needed. I don't think the ConditionalProcessor should be constructed from the script factory at all. Instead, have `maybeWrapConditional` or something like that which takes the processor we have already constructed, and creates/wraps it with a conditional processor (or returns the input if there is no conditional). There is no reason to have a factory for the conditional processor, just parse the config directly in that method, and construct there based on the config.

Reply:
@rjernst you will still need the `ScriptFactory` as a method input to `org.elasticsearch.ingest.ConfigurationUtils#readProcessorConfigs` in some form or another won't you?

I 100% agree, that the current solution looks kind of convoluted :), but it was the only way I saw out of blowing up the change-set with passing the `ScriptFactory` into the methods inside `ConfigurationUtils`.
Looking at your discussion so far it seems it's probably preferable to go with the bigger changeset + cleaner solution I guess? :)
Sorry again about the confusion I caused here :)

Reply:
Yeah, you are right, that method would have to take ScriptFactory. But I think this makes sense? And I think the signature changes should be mostly minimal? Maybe I am underestimating the extent, but I think it is fairly well contained to a handful of methods.

Reply:
When I first tried it, it looked like a fairly massive change :) (but this is also really subjective).
I'll just code it up today and we can take a look, the result is def. nicer than what we have here ... so probably worth it